### PR TITLE
chore: refactor to use testcontainers + bazel

### PR DESCRIPTION
**Pull request by omerlh**
While wanting to add another feature (handling generic consumer errors like we do with 500), I tried running the test:
* gradle took for ever (docker build)
* test are falky

So this PR introduced 2 improvment:
* Replacing gradle with Goolge's Bazel, which is so so so much faster
* Using test containers for testing

### CODE REVIEW

File: tests/specs/tests.spec.ts.bu [hunk 1]
```diff
@@ -0,0 +1,208 @@
+import delay from 'delay';
```
Comment (noamokman) [inline tests/specs/tests.spec.ts.bu:1]: whats this file?

Comment (omerlh) [reply tests/specs/tests.spec.ts.bu:1]: The old tests which I forgot to delete 🤦

File: package.json [hunk 2]
```diff
@@ -14,36 +14,42 @@
...
         "delay": "^4.2.0",
-        "jest": "^24.9.0",
-        "jest-cli": "^24.7.1",
-        "jest-teamcity-reporter": "^0.9.0",
+        "jest": "^29.3.1",
+        "jest-config": "^29.3.1",
         "kafka-node": "4.1.3",
         "lodash": "^4.17.15",
         "node-fetch": "^2.6.0",
         "simple-fake-server-server-client": "^2.0.1",
```
Comment (guysegal) [inline package.json:35]: remove it

**Review by guysegal — COMMENTED** [review]
let's consider this a major change.
If I remember correctly its defined here:https://github.com/osskit/dafka-consumer/pull/42/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R3

File: package.json (hunk 2 above)
Comment (omerlh) [reply package.json:35]: why? we need both I think

**Comment by omerlh** [comment]
What is defined?

File: package.json (hunk 2 above)
Comment (guysegal) [reply package.json:35]: we still use simple-fake-server-server-client after you added @osskit/wiremock-client?

Comment (omerlh) [reply package.json:35]: I didn't saw you talked about it, just removed it

File: bazel-dafka-consumer [hunk 3]
```diff
@@ -0,0 +1 @@
+/private/var/tmp/_bazel_omerlh/647e4a13de7ee4f4bd35fdbdec270a8a/execroot/__main__
```
Comment (SharonGrossman) [inline bazel-dafka-consumer:1]: 🤔

**Comment by omerlh** [comment]
Oopps both files should not be committed, fixed

**Review by guysegal — APPROVED** [review]

**Review by guysegal — APPROVED** [review]

**Pull request merged**